modules/auxiliary/scanner/rsync/modules_list.rb

```diff
@@ -6,64 +6,201 @@
 require 'msf/core'
 
 class Metasploit3 < Msf::Auxiliary
-
   include Msf::Exploit::Remote::Tcp
   include Msf::Auxiliary::Scanner
   include Msf::Auxiliary::Report
 
+  RSYNC_HEADER = '@RSYNCD:'
+
   def initialize
     super(
-      'Name'        => 'Rsync Unauthenticated List Command',
-      'Description' => 'List all (listable) modules from a rsync daemon',
-      'Author'      => 'ikkini',
+      'Name'        => 'List Rsync Modules',
+      'Description' => %q(
+        An rsync module is essentially a directory share.  These modules can
+        optionally be protected by a password.  This module connects to and
+        negotiates with an rsync server, lists the available modules and,
+        optionally, determines if the module requires a password to access.
+      ),
+      'Author'      => [
+        'ikkini', # original metasploit module
+        'Jon Hart <jon_hart[at]rapid7.com>', # improved metasploit module
+        'Nixawk' # improved metasploit module
+      ],
       'References'  =>
         [
           ['URL', 'http://rsync.samba.org/ftp/rsync/rsync.html']
         ],
       'License'     => MSF_LICENSE
     )
+
     register_options(
       [
+        OptBool.new('TEST_AUTHENTICATION',
+                    [ true, 'Test if the rsync module requires authentication', true ]),
         Opt::RPORT(873)
-      ], self.class)
+      ]
+    )
+
+    register_advanced_options(
+      [
+        OptBool.new('SHOW_MOTD',
+                    [ true, 'Show the rsync motd, if found', false ]),
+        OptInt.new('READ_TIMEOUT', [ true, 'Seconds to wait while reading rsync responses', 2 ])
+      ]
+    )
+  end
+
+  def peer
+    "#{rhost}:#{rport}"
+  end
+
+  def read_timeout
+    datastore['READ_TIMEOUT']
+  end
+
+  def rsync_requires_auth?(rmodule)
```

Comment:
Here's my opinion about this method:

1. The method name sort of implies asking for a boolean answer. It sort of does, but then it also return strings... so not really. This creates some confusion.
2. The way this method is used is to store whatever value it returns in a note. So this also kind of implies it wants strings, and not boolean. Maybe?

My suggestion to implement this method, either one of the following:

1. Rename the method. Something that doesn't end with a question mark. Maybe: ```def get_rsync_auth_state```.
2. In this method, return constants. For example, define these constants:

```ruby
AUTH_REQURED = 'Authentication Required'
AUTH_HEADER_OK = 'Header OK'
AUTH_UNEXPECTED = 'Unexpected Response'
AUTH_NO_RESPONSE = 'No Response'
```

And then in the method:

```ruby
```

You don't have to implement the exact same way. This is demonstrating the idea.

Or, if you don't like constants, you could use Symbols. So something like:

```ruby
  def get_rsync_auth_state(rmodule)
    sock.puts("#{rmodule}\n")
    res = sock.get_once(-1, read_timeout)
    if res
      if res =~ /^#{RSYNC_HEADER} AUTHREQD/
        :auth_required
      elsif res =~ /^#{RSYNC_HEADER} OK/
        :auth_header_ok
      else
        vprint_error("#{peer} - unexpected response when connecting to #{rmodule}: #{res}")
        :auth_unexpected
      end
    else
      vprint_error("#{peer} - no response when connecting to #{rmodule}")
      :auth_no_response
    end

  end
```

And then when you call the method, just make sure to do ```.to_s``` to convert the symbol to string.

Here's just my proposal. If you don't think the issue I've brought up is really an issue, then please continue :-)